modules/invoice/view/ajax.php

```diff
@@ -50,4 +50,282 @@
+elseif($_lib['input']->getProperty('action_invoice_detailed_line_control_check')) {
+  $voucher_ids = array();
+  $query_journal_vouchers = "
+    SELECT
+      v.*
+    FROM
+      voucher v
+      JOIN
+      accountplan ap
+      ON
+        v.AccountPlanID = ap.AccountPlanID
+    WHERE
+      -- Exclude the hovedbok lines
+      ap.EnableReskontro = 0 AND
+      v.VoucherType = 'S' AND
+      v.Active = 1 AND
+      v.JournalID = $InvoiceID
+    ";
+  $result_journal_vouchers = $_lib['db']->db_query($query_journal_vouchers);
+  $journal_vouchers = array();
+  while($voucher = $_lib['db']->db_fetch_object($result_journal_vouchers)){
+    $journal_vouchers[$voucher->VoucherID] = $voucher;
+  }
+  $messages = '';
+
+  // Check if total differs
+  $query_line_control_total_differs = "
+    SELECT
+      v.VoucherID,
+      IF(i.TotalCustPrice = v.AmountIn OR i.TotalCustPrice = v.AmountOut, 0, 1) AS TotalDiffers
+    FROM
+      invoiceout i
+      JOIN
+      voucher v
+      ON
+        i.InvoiceID = $InvoiceID AND
+        i.InvoiceID = v.JournalID AND
+        v.VoucherType = 'S' AND
+        v.Active = 1
+      JOIN
+      accountplan ap
+      ON
+        v.AccountPlanID = ap.AccountPlanID AND
+        ap.AccountPlanType = 'customer'
+    ORDER BY
+      VoucherID
+    LIMIT 1
+  ";
+  $line_control_total_differs = $_lib['db']->get_row(
+    array(
+      'query' => $query_line_control_total_differs
+    )
+  );
+  if ($line_control_total_differs->TotalDiffers) {
+    $messages .= "Total differs";
+  } else {
+    // TOTAL Matches
+    array_push($voucher_ids, $line_control_total_differs->VoucherID);
+    unset($journal_vouchers[$line_control_total_differs->VoucherID]);
+  }
+
+  // Check if lines and their VAT differ
+  $lines_differ = false;
+  $vat_lines_differ = false;
+  $query_invoice_lines = "
+    SELECT
+      il.*,
+      ROUND(
+        (
+          SELECT
+            IFNULL(
+              SUM(
+                IF(
+                  ilac.ChargeIndicator = 1,
+                  ilac.Amount,
+                  -ilac.Amount
+                )
+              ),
+              0
+            )
+          FROM
+            invoicelineallowancecharge ilac
+          WHERE
+            ilac.AllowanceChargeType = 'line' AND
+            ilac.InvoiceType = 'out' AND
+            ilac.InvoiceLineID = il.LineID
+        ),
+        2
+      ) AS InvoiceLineAllowanceChargeTotal
+    FROM
+      invoiceoutline il
+    WHERE
+      il.Active = 1 AND
+      il.InvoiceID = $InvoiceID
+    ";
+  $result_invoice_lines = $_lib['db']->db_query($query_invoice_lines);
+  while($invoice_line = $_lib['db']->db_fetch_object($result_invoice_lines)){
+    $line_amount = round($invoice_line->QuantityDelivered * $invoice_line->UnitCustPrice + $invoice_line->InvoiceLineAllowanceChargeTotal, 2);
+    $line_vat_amount = round($invoice_line->Vat/100 * $line_amount, 2);
+    $line_amount = round($line_amount + $line_vat_amount, 2);
+    $query_line = "
```

Comment:
use the same query for line and vat just pass the amount as the thing to change from a placeholder value